docs/environment-and-software/user-environment/spack/spack.md

```diff
@@ -119,4 +119,461 @@ You could also limit the scope of this specific projection to **parallel-netcdf*
+              Activates support for Scalapack
+          superlu-dist [true]         false, true
+              Activates support for superlu-dist (only parallel)
+
+        when +rocm
+          amdgpu_target [none]        none, gfx1010, gfx1011, gfx1012, gfx1013, gfx1030, gfx1031, gfx1032, gfx1033, gfx1034, gfx1035,
+                                      gfx1036, gfx1100, gfx1101, gfx1102, gfx1103, gfx701, gfx801, gfx802, gfx803, gfx900, gfx900:xnack-,
+                                      gfx902, gfx904, gfx906, gfx906:xnack-, gfx908, gfx908:xnack-, gfx909, gfx90a, gfx90a:xnack+,
+                                      gfx90a:xnack-, gfx90c, gfx940, gfx941, gfx942
+              AMD GPU architecture
+
+    Build Dependencies:
+        blas       giflib   hipsolver     kokkos          llvm-amdgpu  mpfr             parmetis   rocrand    scotch        trilinos
+        cgns       gmake    hipsparse     kokkos-kernels  memkind      mpi              parmmg     rocsolver  sowing        valgrind
+        cuda       gmp      hsa-rocr-dev  lapack          metis        mumps            python     rocsparse  strumpack     zlib-api
+        diffutils  hdf5     hwloc         libpng          mkl          netcdf-c         random123  rocthrust  suite-sparse  zoltan
+        exodusii   hip      hypre         libx11          mmg          p4est            rocblas    saws       superlu-dist
+        fftw       hipblas  jpeg          libyaml         moab         parallel-netcdf  rocprim    scalapack  tetgen
+
+    Link Dependencies:
+        blas      gmake      hipsparse     kokkos-kernels  memkind  mpi              parmmg     rocsparse  strumpack     zlib-api
+        cgns      gmp        hsa-rocr-dev  lapack          metis    mumps            random123  rocthrust  suite-sparse  zoltan
+        cuda      hdf5       hwloc         libpng          mkl      netcdf-c         rocblas    saws       superlu-dist
+        exodusii  hip        hypre         libx11          mmg      p4est            rocprim    scalapack  tetgen
+        fftw      hipblas    jpeg          libyaml         moab     parallel-netcdf  rocrand    scotch     trilinos
+        giflib    hipsolver  kokkos        llvm-amdgpu     mpfr     parmetis         rocsolver  sowing     valgrind
+
+    Run Dependencies:
+        None
+
+    Licenses:
+        None
+    ```
+
+**Concretization and installation**
+
+As indicated above, Spack can be *very* complex and often requires iteration to behave as intended - this is certainly the case with PETSc.  Below we walk through some common issues and their resolution.
+
+As a first attempt, we use the `spec` subcommand to inspect the results of Spack's concretization. Inspecting the output of `spack spec -I -l petsc %oneapi@2023.2.1 ^cray-mpich@8.1.27` shows an incompatible mix of compilers are chosen for PETSc and some of its dependencies. (expand the box below for full details).
+??? danger "Simple `spack spec` concretization - *FAILS*"
```

Comment:
IMO, it isn't clear that this is a failed concretization. Looking at this, while I see perhaps one potential issue, nothing really looks too out of the ordinary. I'm assuming that this particular realization fails at install time, but I think calling this a failed concretization is confusing.